rfcs/rdm-0079-collections.md

```diff
@@ -0,0 +1,337 @@
+                                     
+<a name="ctrees">🔑</a> All the collections must be linked to a `CollectionTree` for organization purposes, however the display of the `CollectionTree` `title` on the UI could be optional. In a future iteration one could add a column to control the `CollectionTree` visibility.
+                                     
+<a name="indices">🔑</a> Apart from the usual PK, FK indices, the following indices are to be added to increase the performance of the most common used queries:
+                                     
+1) Index on `Collection.path`, possibly containing the `depth` column to allow matching only subcollections until a certain depth. 
+    This index can initially be a B-TREE, since `LIKE` operations are also supported when they are anchored to the left (see PostgreSQL [documentation](https://www.postgresql.org/docs/current/indexes-types.html#INDEXES-TYPES-BTREE))
+    There are other types of indices that can be used, but B-Trees should be enough for the foreseeable future of collections. If there is a need to perform unanchored queries, e.g. `WHERE path LIKE %foo%` then this has to be reconsidered.
+
+**Collection queries**
+                                     
+Collections, as modelled here, do not store the information of the records that exist under that collection. I.e. it is not possible to know whether record X exists in a collection. Alternatively, collections store the **information on how the records can be queried**.
+                                     
+<a name="queries">🔑</a>  One of the key points of the design of collections, is to answer the question `"What is the query that a collection should execute?"`.
+                                     
+We answer this question by concatenating the ascendants queries. See the following example, where each collection query is defined in blue and the executed query in orange:
+                                     
+![](0079/8.png){width=700}
+
+We have decided this way for (1) consistency and (2) performance:
+                                     
+1.  a record that belongs to a subcollection is guaranteed to also be included in its parent(s) collection(s).
+2.  resolving the query requires `N` primary key lookups on the `collections` table, where `N` is the node depth in the tree (e.g. 2 for `Particles` in the previous example)
+                           
+**Collection endpoints**
+
+We identified two new endpoints:
+
+1. `/<community_slug>/collections`
+2. `/<community_slug>/collections/<collection_slug>`
+
+The endpoint for a non-community collection is not yet defined, and will be listed as an open question in this document.
+
+### Configuration
+
+Initially collections entry point in the UI will be toggleable, e.g. disabled or enabled. However, this will not avoid the creation of the Database tables.
```

Comment:
should this be a module then? if needed we install it and then it adds the tables?

Reply:
Personally, that was my first thought. But since collections are tightly linked to records (and communities), we will add them in `invenio-rdm-records`. So to answer your question, one can't opt-out from having the tables